athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```diff
@@ -0,0 +1,385 @@
+
+        Set<Split> splits = new HashSet<>();
+        FieldReader fieldReader = request.getPartitions().getFieldReader("preparedStmt");
+
+        String sqlStatement  = fieldReader.readText().toString();
+        String catalogName = request.getCatalogName();
+
+        /**
+         * Build sql based on what query has been issued and concat it with the export to parquet command
+         */
+
+        try{
+            Connection connection = getOrCreateConn(request);
+
+            logger.info("Generated EXPORT statement is: " + sqlStatement);
+
+            //get AWS KEYS from AWS Secrets Manager
+            String awsSecrets = getSecret(System.getenv(AWS_SECRETS));
+
+            //extracting Secrets
+            String replacedSecretString = (awsSecrets.replace("{","")
+                    .replace("}","")
+                    .replace("\"","")
+                    .replaceAll("\\s+",""));
+
+            Map<String, String> result = Arrays.stream(replacedSecretString.split(","))
+                    .map(k ->  k.split(":"))
+                    .collect(Collectors.toMap(
+                            a -> a[0],  //key
+                            a -> a[1]   //value
+                    ));
+
+            String awsSecretKey = result.get("aws_secret_access_key");
+            String awsAccessId = result.get("aws_access_key_id");
+
+            //Generating the sql to set the AWS Session Config on Vertica
+            StringBuilder sb = new StringBuilder();
+            ST temp = new ST("ALTER SESSION SET AWSAuth='<access_key>:<secret_key>'");
+            temp.add("access_key", awsAccessId);
+            temp.add("secret_key", awsSecretKey);
+            String credS = temp.render();
+
+            PreparedStatement credPS = connection.prepareStatement(credS);
+            PreparedStatement exportSQL = connection.prepareStatement(sqlStatement);
+
+            // Set the AWS Region on Vertica
+            String defaultRegion = System.getenv(DEFAULT_REGION).toLowerCase();
+            String awsRegion =  "SELECT SET_CONFIG_PARAMETER('AWSRegion','"+defaultRegion+"') AS REGION_NAME";
+
+            PreparedStatement setAwsRegion = connection.prepareStatement(awsRegion);
+
+            //execute the query to set credentials
+            credPS.execute();
+
+            //execute the query to set Region
+            ResultSet rSet = setAwsRegion.executeQuery();
+
+            //execute the query to export the data to S3
+            exportSQL.execute();
+
+        } catch (SQLException e) {
+            logger.info("Exception in doGetSplits executing the Prepared Statement " + e.getMessage());
+        }
+
+        //generating the file name
+        String exportBucket = System.getenv("export_bucket");
+        String s3File = "s3://" + exportBucket + "/" + request.getQueryId().replace("-","") + ".parquet";
+
+        //creating the split
+        Split split = Split.newBuilder(makeSpillLocation(request), makeEncryptionKey())
```

Comment:
looks like we are only exporting to a single file? Is this our first iteration and we are planning to do the multi-file export for better parallelism later?

Reply:
This was a left over code I forgot to delete. I do not use this file anywhere in the program flow. Removed this.